app/_components/companion-recruitment-filter.tsx

```diff
@@ -35,7 +40,38 @@ const CompanionRecruitmentFilter = () => {
   const ageRef = useRef([20, 30]);
   const personCountRef = useRef(1);
 
-  const handle = () => {};
+  const joinQuery = (...targets: string[][]) => {
+    return targets.map(target => target.join('&'));
+  };
+
+  const setObject = (regionQuery: string) => {
+    return {
+      gender: radioRef.current,
+      regions: regionQuery,
+      transportation: transportationRef.current,
+      startAge: ageRef.current[0],
+      endAge: ageRef.current[1],
+      totalPeople: personCountRef.current,
+    };
+  };
+
+  const getQuery = () => {
+    const [regionQuery] = joinQuery(
+      Object.keys(checkboxRef.current).filter(key => checkboxRef.current[key]),
+    );
+
+    const { gender, regions, startAge, endAge, transportation, totalPeople } =
+      setObject(regionQuery);
+
+    return `gender=${gender}&region=${regions}&startAge=${startAge}&endAge=${endAge}&transportation=${transportation}&totalPeople=${totalPeople}`;
```

Comment:
이부분도 반복되는것 같아 함수로 만들수 있을것 같은데 혹시 어떻게 생각하시나요?!🧐

Reply:
```suggestion
    const objectToQueryString = (object) => Object.entries(object).reduce(acc,[key,vlaue])=>`${acc}&${key}=${value}`,''};
    const newObject = setObject(regionQuery);
    return objectToQueryString(newObject);
```

이런식으로 만들어봤는데.. 그대로 명시해주는것도 나을것 같다는생각이 드네요..🫠

Reply:
오 👍👍 나중에 필드가 추가되거나 삭제되더라도 수정하지 않을 수 있을 것 같아요